WooCommerce/Classes/POS/Analytics/WooAnalyticsEvent+PointOfSale.swift

```diff
@@ -77,81 +115,121 @@ extension WooAnalyticsEvent {
+            )
         }
 
         static func pointOfSaleSearchRemoteResultsFetched(itemType: POSItemType,
                                                           resultsCount: Int,
                                                           millisecondsSinceRequestSent: Int) -> WooAnalyticsEvent {
             WooAnalyticsEvent(statName: .pointOfSaleSearchRemoteResultsFetched,
                               properties: [
-                                Key.itemListType: itemType.analyticsValue,
                                 Key.resultsCount: "\(resultsCount)",
                                 Key.millisecondsSinceRequestSent: "\(millisecondsSinceRequestSent)"
                               ])
         }
     }
 }
 
-private extension WooAnalyticsEvent.PointOfSale.CartItemType {
-    var analyticsValue: String {
-        switch self {
-        case .simpleProduct:
-            return "simple"
-        case .variation:
-            return "variation"
+extension WooAnalyticsEvent.PointOfSale {
+    enum Source: String {
+        case product
+        case variation
+        case coupon
+        case cart
+        case error
+
+        init(itemType: POSItemType) {
+            switch itemType {
+            case .product:
+                self = .product
+            case .variation:
+                self = .variation
+            case .coupon:
+                self = .coupon
+            }
+        }
+
+        init(itemListType: ItemListType) {
+            switch itemListType {
+            case .products:
+                self = .product
+            case .coupons:
+                self = .coupon
+            }
         }
     }
-}
 
-private extension ItemListType {
-    var analyticsValue: String {
-        switch self {
-        case .products:
-            return "products"
-        case .coupons:
-            return "coupons"
+    enum SourceType: String {
+        case list
+        case search
+        case preSearch = "pre_search"
+
+        init(isSearching: Bool, searchTerm: String = "") {
+            switch (isSearching, searchTerm.isEmpty) {
+            case (false, _):
+                self = .list
+            case (true, true):
+                self = .preSearch
+            case (true, false):
+                self = .search
+            }
         }
     }
 
-    var addItemSourceAnalyticsValue: String {
-        switch self {
-        case .products(search: true):
-            return "search_result"
-        case .products(search: false),
-                .coupons:
-            return "list"
-        }
+    enum ItemType: String {
+        case product
+        case coupon
     }
-}
 
-private extension POSItemType {
-    var analyticsValue: String {
-        switch self {
-        case .product:
-            return "product"
-        case .variation:
-            return "variation"
-        case .coupon:
-            return "coupon"
+    enum ProductType: String {
```

Comment:
nit: I think "product type" usually refers to the type of the product, so I was expecting "variable" for the "variation" case. Is it perhaps referring to the cart item type?

Reply:
Good question. You're likely correct, the way of thinking was that `variable` product is selected to open a list of variations, and `varation` is selected to add to the cart. We synced this with Android. Do you think it's problematic to leave it like that?

Reply:
nit: then maybe we can change the name of the enum to refer to the cart item type? `ProductType` enum in Yosemite has a different meaning, and it could be confusing having two enums of the same name in the codebase with different meanings.